### ISSUE TITLE
Custom keybindings for Wordplay syntax

### ISSUE THREAD

**Issue by melissachen2000** [wordplaydev/wordplay#826]
### What are you trying to do that you can't?

In a co-design session with 8th grade students and their teacher, students pointed out challenges with writing Wordplay code. An important piece of context is that it seems like most, if not all, participants have prior experience, and these challenges arose from conversation about how it is difficult to translate concepts and syntax from other languages to Wordplay's concepts and syntax. The examples they gave:

1. Some of the symbols that are frequently used in Wordplay's syntax are not available on their keyboards
2. Issues with using and copying emojis [Melissa's note: I haven't been able to replicate this issue with emojis, custom characters, or reserved words (e.g., using 💬 instead of `Phrase`), maybe it was fixed?]
3. They expected that the tab key could be used to indent their code but it does not. This would typically work in other languages they seemed familiar with (e.g., Python)

### What is your idea?

Students suggested two things:

1. Recognizing that the syntax design choices were made to support multilingualism and not privilege any particular writing system over another, students suggested to instead have user-specific custom keybindings. For example, they can tie the tab key to the insert tab character in Wordplay, which improves their own experience without impacting others. 
2. See #825 -- community documentation can help students make connections between Wordplay and other languages.

**Comment by amyjko** [comment]
This design problem intersects with a lot of existing functionality:

* There are key bindings already for many of the language's symbols. I wonder if they aren't discoverable enough; they are shown in tooltips and in the keyboard shortcut feature, but perhaps those are hard to find.
* Tab can't be used without violating WCAG 2.1 AA requirements, but we could make a feature that supports that. It's described in #758.

Let's scope this issue to keyboard shortcut customization and discoverability. I think this is a relatively advanced feature, and probably not as critical path as other requests, so let's keep it but not put it in the milestone.

**Comment by amyjko** [comment]
A few notes on this one:

* We have tab behavior customization already now
* We should fix the window picker switcher collision even if we don't implement broad keybindings support
* Emoji copying should be fixed now

So the smallest this would be is fixing any cross-platform shortcut collisions and the largest would be general support for modifying keybindings (which may generally be a good accessibility feature).
